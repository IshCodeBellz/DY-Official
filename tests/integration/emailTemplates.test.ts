import {
  buildOrderConfirmationHtml,
  buildPaymentReceiptHtml,
} from "@/lib/server/mailer";

describe("email templates", () => {
  const baseOrder: any = {
    id: "order_12345",
    totalCents: 2599,
  };

  test("order confirmation html snapshot", () => {
    const html = buildOrderConfirmationHtml(baseOrder);
    expect(html).toMatchSnapshot();
  });

  test("payment receipt html snapshot", () => {
    const html = buildPaymentReceiptHtml(baseOrder);
    expect(html).toMatchSnapshot();
  });
});
