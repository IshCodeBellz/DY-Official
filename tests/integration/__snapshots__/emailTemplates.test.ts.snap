// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`email templates order confirmation html snapshot 1`] = `
"<!doctype html><html><body style="font-family:Arial,sans-serif;line-height:1.4;color:#222;">
  <h1 style="font-size:20px;margin:0 0 12px;">Thanks for your order!</h1>
  <p style="margin:0 0 12px;">Order <strong>#order_12345</strong> has been received.</p>
  <p style="margin:0 0 12px;">Total: <strong>$25.99</strong></p>
  <p style="font-size:12px;color:#666;">You will receive another email when payment is confirmed.</p>
  </body></html>"
`;

exports[`email templates payment receipt html snapshot 1`] = `
"<!doctype html><html><body style="font-family:Arial,sans-serif;line-height:1.4;color:#222;">
  <h1 style="font-size:20px;margin:0 0 12px;">Payment received</h1>
  <p style="margin:0 0 12px;">We've captured payment for order <strong>#order_12345</strong>.</p>
  <p style="margin:0 0 12px;">Amount: <strong>$25.99</strong></p>
  <p style="font-size:12px;color:#666;">We'll start fulfilling your order shortly.</p>
  </body></html>"
`;
